plugin/src/main/kotlin/tmg/gradle/plugin/gms/location/TmgLocationFixPlugin.kt

```diff
@@ -0,0 +1,54 @@
+package tmg.gradle.plugin.gms.location
+
+import com.android.build.api.instrumentation.FramesComputationMode
+import com.android.build.api.instrumentation.InstrumentationScope
+import com.android.build.api.variant.AndroidComponentsExtension
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+@Suppress("unused")
+public abstract class TmgLocationFixPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        project.plugins.withId("com.android.library") {
+            project.logger.warn("[TMG] tmg.gms.location.fix plugin is most effective when applied to the Application module")
+        }
+
+        val ext = project.extensions.create("tmgLocationFix", TmgLocationFixExtension::class.java)
+
+        project.extensions.configure(AndroidComponentsExtension::class.java) { androidComponents ->
+            @Suppress("UnstableApiUsage")
+            androidComponents.finalizeDsl {
+                val selector = ext.variantSelector.getOrElse(androidComponents.selector().all())
+                androidComponents.onVariants(selector) { variant ->
```

Comment:
we may be able to replace this with `androidComponents.registerExtension`, to allow the app to configure the plugin per-variant. I.e., to enable strict mode on release builds, or enable the rewriting behavior only on debug builds, to replace ext.variantSelector